lib/cache/sfAPCCache.class.php

```diff
@@ -32,9 +32,15 @@ class sfAPCCache extends sfCache
    */
   public function initialize($options = array())
   {
-    parent::initialize($options);
+    // Check if the APC is enabled and the apc_store function is available
+    // Be aware that php7's apcu extension removes the  backward compatibility with the original
+    // apcu used in php 5.x. The "apcu-bc" extension to be enabled to provide backward compatibility.
+    if (!function_exists('apc_store') || !ini_get('apc.enabled'))
+    {
+      throw new sfConfigurationException('sfAPCCache class needs the "apc" extension and the "apc_store" function to be enabled.');
```

Comment:
Throwing an exception here is a BC breaks. Then to provide a BC layer I propose to introduce an option for that.